src/manifest.rs

```diff
@@ -572,12 +571,15 @@ fn search(dir: &Path) -> Result<PathBuf> {
     }
 }
 
-fn get_version(old_dep: &toml_edit::Item) -> Result<toml_edit::Item> {
-    if str_or_1_len_table(old_dep) {
-        Ok(old_dep.clone())
+fn get_version(old_dep: &toml_edit::Item) -> Result<&str> {
+    if let Some(req) = old_dep.as_str() {
+        Ok(req)
     } else if old_dep.is_table_like() {
```

Comment:
This actually fixes a bug.

When you have a `1_len` table, we are returning the table, rather than doing `old_dep.get("version")`.  I'm surprised this hasn't been caught in tests.